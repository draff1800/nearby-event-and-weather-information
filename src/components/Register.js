import React, { Component } from 'react';
import { Alert, Button, Image, KeyboardAvoidingView, Text, TextInput, ToastAndroid, View } from 'react-native';
import { styles } from '../Styles';
import FirebaseConfig from "../../config/Firebase";


export default class Login extends Component {
  static navigationOptions = {
    title: 'Register',
  };

  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      confirmPassword: '',
    };
  }

  handleRegister = () => {
    if (this.state.password !== this.state.confirmPassword) {
      Alert.alert('Oops!', 'Passwords do not match.')
    } else {
      FirebaseConfig.auth()
        .createUserWithEmailAndPassword(this.state.email, this.state.password)
        .then(() => ToastAndroid.show('Successfully registered!', ToastAndroid.SHORT))
        .then(() => this.props.navigation.navigate('Login'))
        .catch(error => Alert.alert('Oops!', error.message))
    }
  };

  render() {
    const {navigate} = this.props.navigation;
    return (
      <View style={styles.container}>
        <KeyboardAvoidingView style={styles.container} behavior="padding" enabled>
          <Image source={{uri: 'https://i.imgur.com/D03aDTg.png'}} style={styles.logoImage}/>
          <Text style={styles.logoText}>Nearby Event and Weather Information</Text>
          <TextInput
            value={this.state.email}
            onChangeText={(email) => this.setState({ email })}
            placeholder={'Email'}
            style={styles.input}
          />
          <TextInput
            value={this.state.password}
            onChangeText={(password) => this.setState({ password })}
            placeholder={'Password'}
            secureTextEntry={true}
            style={styles.input}
          />
          <TextInput
            value={this.state.confirmPassword}
            onChangeText={(confirmPassword) => this.setState({ confirmPassword })}
            placeholder={'Confirm password'}
            secureTextEntry={true}
            style={styles.input}
          />

          <View styles={styles.buttonContainer}>
            <Button
              title={'Register'}
              onPress={this.handleRegister}
            />
          </View>
          <View style={styles.buttonContainer}>
            <Button
              title={'Login'}
              onPress={() => navigate('Login')}
            />
          </View>
        </KeyboardAvoidingView>
      </View>
    );
  }
}
