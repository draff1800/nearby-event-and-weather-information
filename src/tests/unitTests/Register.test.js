import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Register from '../../Components/Register';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const navigation = {navigate: jest.fn()};
const tree = renderer.create(<Register navigation={navigation}/>).toJSON();
const wrapper = shallow(<Register navigation={navigation}/>);

describe('<Register />', () => {
  it('has 1 child - KeyboardAvoidingView', () => {
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should have a logo, title, three input fields and login / register buttons', () => {
    expect(wrapper.find('Image')).toHaveLength(1);
    expect(wrapper.find('Text')).toHaveLength(1);
    expect(wrapper.find('TextInput')).toHaveLength(3);
    expect(wrapper.find('Button')).toHaveLength(2);
  });
});
