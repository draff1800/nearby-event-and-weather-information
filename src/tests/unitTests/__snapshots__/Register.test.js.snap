// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.imgur.com/D03aDTg.png",
        }
      }
      style={
        Object {
          "height": 100,
          "resizeMode": "contain",
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      Nearby Event and Weather Information
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 44,
          "marginBottom": 10,
          "padding": 10,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 44,
          "marginBottom": 10,
          "padding": 10,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Confirm password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 44,
          "marginBottom": 10,
          "padding": 10,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      styles={
        Object {
          "padding": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Register
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
